enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,398 @@
+work, [RHCOS will include the ability to run a text-based UI to configure
+networking](https://github.com/openshift/enhancements/pull/291) in the ramdisk:
+`nmtui`.  This installation process also provides the ability to provide
+networking configuration via Ignition if the administrator does not want to
+follow an interactive configuration process.
+
+These enhancements provide some excellent improvements to early network
+configuration capabilities.  However, it is currently most beneficial to the
+UPI bare metal flow.  The IPI bare metal flow will need to do more work to
+figure out how to make use of these tools.  Bare Metal IPI makes use of a
+different deployment ramdisk that is part of the Ironic project.  While
+conceptually similar, it has some significant differences in implementation.
+Future integration work may provide better alignment between the Bare Metal IPI
+tooling and the RHCOS installer.
+
+### SR-IOV Operator
+
+Some clusters may want to allocate SR-IOV devices to workloads for a secondary
+network interface with high performance.  These devices are managed by the
+[SR-IOV Operator[(https://github.com/openshift/sriov-network-operator).
+
+
+## Future Improvements
+
+### Disabling initramfs Networking
+
+Configuration of initramfs networking is lacking in some desired flexibility,
+as shown by the discussion in [this ignition
+issue](https://github.com/coreos/ignition/issues/979).  The default behavior in
+RHCOS is to attempt to get a DHCPv4 or DHCPv6 lease on one of the host’s
+interfaces.  This can cause a lot of undesirable delay if there are a lot of
+network interfaces present and DHCP is not available.  It will also cause
+booting to fail if DHCP never succeeds.
+
+One improvement is the ability to disable initramfs networking completely,
+which would be useful in combined with providing the host a full ignition
+configuration without needing to retrieve the bulk of it from a network
+location.  See “Flattened Ignition Configuration” for some more info on that
+piece.
+
+### NetworkManager in initramfs
+
+Fedora CoreOS has already moved to running NetworkManager in the initramfs.
+More info can be found in
+<https://github.com/coreos/fedora-coreos-tracker/issues/394>.
+
+### MCO Managed Pointer Ignition Configuration
+
+There is a WIP proposal to moving management of the “pointer” configuration to
+the MCO, such that e.g upgrade of the ignition version would be possible
+without manual fixes, ref
+<https://github.com/openshift/machine-config-operator/pull/1792>
+
+### Flattened Ignition Configuration
+
+The per-role `MachineConfigPool` resources result in a `MachineConfig` resource
+that is rendered by the MCO, and exposed via the MCS - this is the
+configuration that’s referenced by the “pointer”  ignition resource.
+
+One reason for this “pointer” ignition approach is that on many cloud platforms
+the `user_data` payload size limit is very low - this means the (potentially
```

Comment:
not clear what this is referring to

Reply:
I tried to rework this to explain better.  Can you take a look and see if it makes sense to you now?

Reply:
yes, better now